ludwig/datasets/loaders/hm_fashion_recommendations.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import List
+from unittest.mock import Mock
+
+import numpy as np
+import pandas as pd
+
+from ludwig.constants import SPLIT
+from ludwig.data.dataframe.pandas import PandasEngine
+from ludwig.data.split import get_splitter
+from ludwig.datasets.loaders.dataset_loader import DatasetLoader
+from ludwig.datasets.loaders.utils import negative_sample
+
+
+def _merge_dataframes(transactions_df, articles_df, customers_df):
+    """Merge the transactions, articles, and customers dataframes into a single dataframe."""
+    # Merge the transactions and articles dataframes
+    transactions_df = pd.merge(
+        transactions_df,
+        articles_df,
+        how="left",
+        left_on="article_id",
+        right_on="article_id",
+    )
+
+    # Merge the transactions and customers dataframes
+    transactions_df = pd.merge(
+        transactions_df,
+        customers_df,
+        how="left",
+        left_on="customer_id",
+        right_on="customer_id",
+    )
+
+    return transactions_df
+
+
+def _split(df):
+    """Split the dataframe into train, validation, and test dataframes.
+
+    The split is done in a chronological manner based on the year_month column. The split is done by customer_id,
+    so that interactions for a given customer are present in all splits.
+
+    Params:
+        df: The dataframe to split.
+
+    Returns:
+        A tuple of (train_df, validation_df, test_df).
+    """
+    backend = Mock()
+    backend.df_engine = PandasEngine(_use_ray=False)
+
+    splitter = get_splitter("datetime", column="year_month", probabilities=(0.7, 0.2, 0.1))
+
+    if not isinstance(df, pd.DataFrame):
+        df = df.compute()
+
+    train_dfs, val_dfs, test_dfs = [], [], []
+    for customer_id in df["customer_id"].unique():
+        # Split per customer_id to ensure that interactions for a customer are across all splits
+        train_df, val_df, test_df = splitter.split(df[df["customer_id"] == customer_id], backend=backend)
+
+        train_dfs.append(train_df)
+        val_dfs.append(val_df)
+        test_dfs.append(test_df)
+
+    return pd.concat(train_dfs), pd.concat(val_dfs), pd.concat(test_dfs)
+
+
+class HMLoader(DatasetLoader):
+    def load_unprocessed_dataframe(self, file_paths: List[str], sample=True) -> pd.DataFrame:
```

Comment:
Need doc string for sample, not obvious what it does or why its default is True.  It looks like it sub-samples the dataset and chooses 100 unique customers?